meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,186 @@
+  - A single term is now limited to 150 possible typo matches for queries containing 1 typo, and to 50 possibilities for queries containing 2 typos
+  - Both single word or multi word queries consider a maximum of 50 synonyms
+  - The total number of words for all synonyms of a single term cannot exceed 100
+  - Queries can now contain a maximum of 1000 words. This change only impacts queries containing very long phrases. Search terms (words not inside a phrase, and individual phrases) remain limited to 10 per query
+- Geo search improvements:
+  - Increased performance when geosorting small sets of documents
+  - Descending sort is now as performant as ascending sort
+- Address assorted minor performance issues: https://github.com/meilisearch/meilisearch/issues/3124, https://github.com/meilisearch/meilisearch/issues/3378, https://github.com/meilisearch/meilisearch/issues/2202, https://github.com/meilisearch/meilisearch/issues/3123
+
+### Relevancy improvements
+
+- The `exactness` ranking rule no longer treats synonyms as exact matches. This boosts documents containing the query exactly as typed by the user. Solves https://github.com/meilisearch/meilisearch/issues/3410
+- Meilisearch will always sort results as if the `words` ranking rule were present with a higher priority than the `attributes`, `exactness`, `typo` and `proximity` ranking rules. This happens even if the `words` ranking rule has been removed or set with a lower priority. Search behavior when using the specified ranking rules before or without `words` was confusing and hurt relevancy
+- Split words are now also treated as possible digrams. This means the query `whit ehorse` may match documents containing `white horse`, which improves the relevancy of typo tolerance
+- N-grams and split words are now ranked lower in comparison to the exact word when processing the `typo` ranking rule. For example, a query for `sun flower` will now return a document titled `Sun Flower` before one titled `Sunflower`
+- Ranking rule behavior no longer depends on the number of ranked documents, improving consistency in terms of both relevancy and latency. Solves https://github.com/meilisearch/meilisearch/issues/3356
+
+## Automated task deletion
+
+Maximum number of tasks the task queue can hold now limited to 1M. Fixes issues with enqueuing new tasks when the database is full.
+
+Once the task queue reaches this limit, Meilisearch will try to delete the oldest 100k tasks. If not all 100k tasks can be deleted due to some of them being unfinished, we will delete as many tasks as possible.
+
+You’ll still be able to register new tasks even after reaching this limit. The engine will only stop you from adding new tasks once the hard limit of 10GiB of tasks is reached (that’s between 5M and 15M of tasks depending on your workflow).
+
+If automated deletion fails to delete anything, Meilisearch will log a warning notifying the user the engine is not working properly and no tasks were enqueued.
```

Comment:
[As indicated in a comment on the docs repo](https://github.com/meilisearch/documentation/issues/2420#issuecomment-1543770321), tasks are still enqueued if Meilisearch fails to delete any tasks.

```suggestion
Maximum number of tasks the task queue can hold now limited to 1M. Fixes issues with enqueuing new tasks when the database is full.

Once the task queue reaches this limit, Meilisearch will try to delete the oldest 100k tasks. If not all 100k tasks can be deleted due to some of them being unfinished, Meilisearch will delete as many tasks as possible. If there are no unfinished tasks, Meilisearch will not delete anything but still enqueue new tasks as usual.

The engine will stop you from adding new tasks once the hard limit of 10GiB of tasks is reached (that’s between 5M and 15M of tasks depending on your workflow). In this case, Meilisearch will try to automatically delete unfinished tasks. If this is unsuccessful, it will not enqueue new tasks, and log a warning notifying the user the engine is not working properly.
```

Reply:
@irevoire, you might want to double check this section of the changelog just to be sure everything's correct.

Reply:
You are right, seems ok to me, but I can be confused regarding some parts